certbot-dns-linode/certbot_dns_linode/dns_linode.py

```diff
@@ -52,19 +56,44 @@ def _cleanup(self, domain, validation_name, validation):
         self._get_linode_client().del_txt_record(domain, validation_name, validation)
 
     def _get_linode_client(self):
-        return _LinodeLexiconClient(self.credentials.conf('key'))
+        api_key = self.credentials.conf('key')
+        api_version = self.credentials.conf('version')
+
+        if not api_version:
+            api_version = 3
+
+            # Match for v4 api key
```

Comment:
It seems like the main difference between v3 and v4 API keys is that v3 API keys are from a larger character set (`[a-zA-Z0-9]`). Thus, a v3 API key _could_ contain only numbers and lower case characters and match this regular expression.

What happens if a user who is unlucky enough to have a v3 API key that _looks like_ a v4 API key upgrades Certbot and tries to renew? It seems like this will fail.

For backwards compatibility (and predictability), would it be reasonable to _require_ that the API version be specified for API version 4? It's slightly less convenient for people creating credentials files for v4, but this would allow us to always use v3 if the API version is not set — which may be safer.

(If we make this change, there will also be some documentation changes required.)

Reply:
Since I thought the chances of accidentally getting a v3 key that looks like a v4 key is quite low, I used the v3 as default and checked it for v4 key.
If a user does indeed get a v3 key that look likes a v4 key, only they'll have to add the override. Everyone else can not worry about extra configuration.

Reply:
On second thought, the odds here seem... safe.

(More-likely-that-I'll-get-hit-by-a-meteor-tomorrow levels of safety, I think... `((10+6)/(10+26+26))^64 = 0.00000000000000000000000000000000000224187174%`.)